src/main/java/no/finn/unleash/DefaultUnleash.java

```diff
@@ -73,12 +77,33 @@ public boolean isEnabled(final String toggleName, final boolean defaultSetting)
         return isEnabled(toggleName, contextProvider.getContext(), defaultSetting);
     }
 
+    public boolean isEnabled(final String toggleName, final FallbackAction fallbackAction) {
+        FeatureToggle featureToggle = toggleRepository.getToggle(toggleName);
+        boolean enabled = isEnabled(featureToggle, contextProvider.getContext(), false);
+
+        if(!enabled && fallbackAction != null) {
+            fallbackAction.apply(toggleName, contextProvider.getContext());
+        }
+
+        return enabled;
+    }
+
+    public boolean isEnabled(final String toggleName, boolean defaultSetting, final FallbackAction fallbackAction) {
+        FeatureToggle featureToggle = toggleRepository.getToggle(toggleName);
+
+        if(featureToggle == null && fallbackAction != null) {
+            fallbackAction.apply(toggleName, contextProvider.getContext());
+        }
+
+        return isEnabled(featureToggle, contextProvider.getContext(), defaultSetting);
```

Comment:
I would rather invert it and have this function being the master and have the other `isEnabled` calls send in a proper fallback-action defined.